plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java

```diff
@@ -71,37 +98,39 @@
       logger.log(Level.WARNING, "Unable to obtain jst.web facet version", ex);
     }
 
-    try {
-      String[] apiLibraryIds = getApiLibraryIds(webFacetVersion);
-      ListenableFuture<IClasspathEntry[]> apiEntries =
-          resolverService.resolveLibraryAttachSources(apiLibraryIds);
-      // may return immediately if cached or locally available
-      if (apiEntries.isDone()) {
-        return apiEntries.get();
-      }
-      Futures.addCallback(
-          apiEntries,
-          new FutureCallback<IClasspathEntry[]>() {
-            @Override
-            public void onSuccess(IClasspathEntry[] entries) {
-              requestClasspathContainerUpdate(project, runtime, entries);
-            }
+    IClasspathEntry[] entries = libraryEntries.getIfPresent(webFacetVersion);
+    if (entries != null) {
+      return entries;
+    }
 
-            @Override
-            public void onFailure(Throwable t) {
-              logger.log(Level.WARNING, "Failed to resolve servlet APIs", t);
+    final IProjectFacetVersion dynamicWebFacetVersion = webFacetVersion;
+    // $NON-NLS-1$ since it's a system job
+    Job resolveJob = new Job("Resolving libraries for " + webFacetVersion) { // $NON-NLS-1$
```

Comment:
Sorry, I actually thought `PluggableJob` was nice here, but since you have `waitForJobs()` now, it may be good with the plain `Job`.

Anyway, is the indentation below off?